app/javascript/components/bookmarks/BookmarkManager.jsx

```diff
@@ -15,15 +13,17 @@ import { log } from '~/lib/metrics-api'
  * Component to bookmark views in Explore tab, as well as existing link/reset buttons
  *
  * @param bookmarks {Array} existing user bookmark objects
- * @param clearExploreParams
+ * @param studyAccession {String} currently loaded study, if present
+ * @param eagerLoad {Boolean} load bookmarks from server on initialization
```

Comment:
Can you help me see why `eagerLoad` is true on the Home page, and false on the Study Overview page?

Reply:
On the Study Overview, we get the bookmarks back from the `study_user_info` API call.  The `BookmarkManager` component isn't rendered until this comes back.  However, there's no analogous API call on the home page, so I added in the explicit call to load the bookmarks when the component is initialized.  This allows the state to set correctly here (e.g. showing if a query has already been bookmarked) without the user having to open the bookmarks modal first.

Reply:
Much appreciated, this is clear to me now.